lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb

```diff
@@ -0,0 +1,139 @@
+require 'fileutils'
+
+Puppet::Type.type(:bootstrap_rpm).provide(:bootstrap_rpm) do
+
+  commands :rpmbuild => 'rpmbuild'
+
+  def create
+    base_dir
+    copy_sources
+    write_specfile
+    build_rpm
+    copy_rpm
+    link_rpm
+  ensure
+    FileUtils.remove_dir(base_dir) if File.directory?(base_dir)
+  end
+
+  def delete
+  end
+
+  def exists?
+    File.exist?(resource[:symlink]) if resource[:symlink]
+  end
+
+	private
+
+  def base_dir
+    @base_dir ||= Dir.mktmpdir
+  end
+
+  def sources_dir
+    mkdir('SOURCES')
+  end
+
+  def build_dir
+    mkdir('BUILD')
+  end
+
+  def spec_dir
+    mkdir('SPECS')
+  end
+
+  def rpm_dir
+    mkdir('RPMS')
+  end
+
+  def copy_sources
+    FileUtils.copy(resource[:script], "#{sources_dir}/#{File.basename(resource[:script])}")
+  end
+
+  def write_specfile
+    File.open("#{spec_dir}/#{resource[:name]}.spec", 'w') do |file|
+      file.write(spec)
+    end
+  end
+
+  def release
+    if !latest_rpm.nil? && File.exist?(latest_rpm)
+      rel = `rpm -qp #{latest_rpm} --queryformat=%{release}`
+      rel.to_i + 1
+    else
+      1
+    end
+  end
+
+  def build_rpm
+    command = ['rpmbuild', '-ba']
+    command += ["#{spec_dir}/#{resource[:name]}.spec"]
+    command += ['--define', "'_topdir #{base_dir}'"]
+    command += ['--define', "'name #{resource[:name]}'"]
+    command += ['--define', "'release #{release}'"]
+
+    output = `#{command.join(' ')}`
+    Puppet.debug(output)
+  end
+
+  def copy_rpm
+    Dir.glob("#{rpm_dir}/noarch/*.rpm") do |file|
+      FileUtils.copy(file, resource[:dest])
+    end
+  end
+
+  def link_rpm
+    FileUtils.ln_s(latest_rpm, resource[:symlink], force: true) if resource[:symlink]
+  end
+
+  def latest_rpm
+    latest = Dir.glob("#{resource[:dest]}/*.rpm").sort
+    false if latest.nil?
+    latest[0]
+  end
+
+  def mkdir(dir)
+    path = File.join(base_dir, dir)
+    Dir.mkdir(path) unless File.exist?(path)
+    path
+  end
+
+  def spec
+    <<~HEREDOC
+      Name:      %{name}
+      Version:   1.0
+      Release:   %{release}
+      Summary:   Subscription-manager consumer certificate for Katello instance %{name}
+
+      Group:     Applications/System
+      License:   GPL
+      Source0:   katello-rhsm-consumer
+      BuildArch: noarch
+
+      Requires:  subscription-manager
+
+      %description
+      Consumer certificate and post installation script that configures rhsm.
+
+      %prep
+
+      %build
+
+      %install
+      rm -rf %{buildroot}
+
+      install -d -m0755 %{buildroot}%{_bindir}
+      install -Dp -m0755 %{SOURCE0} %{buildroot}%{_bindir}/katello-rhsm-consumer
+
+      %clean
+      rm -rf %{buildroot}
+
+      %post
+      /bin/bash /usr/bin/katello-rhsm-consumer
+
+      %postun
+      test -f /etc/rhsm/rhsm.conf.kat-backup && command cp /etc/rhsm/rhsm.conf.kat-backup /etc/rhsm/rhsm.conf
```

Comment:
Isn't this introducing a bug again?
https://github.com/theforeman/puppet-certs/commit/7bea39e9a03a5a34c675174f9e43b2bc44de51f0